### PR TITLE
feat: preview step

### LINKED ISSUE #65: create the third step of app configuration (Preview app)
image

### PR DESCRIPTION
**Pull request by LinaYahya**
closes #65

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const debounce = (func: Function, delay: number) => {
+  let timer: ReturnType<typeof setTimeout>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (...args: any[]) => {
```
Comment (pyphilia) [inline src/utils/index.ts:5]: Not sure it works.
```suggestion
  return (...args: unknown[]) => {
```

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the PR! It's not really clear how it works however, if you could explain what are labels and how you "compute" the key components such as "choices" and "to be filled boxes".
Also for image dimensions I'm not sure it's a good idea to save it in settings.

File: src/modules/builder/configuration/AddImageStep.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ const AddImageStep = ({ moveToNextStep }: Props): JSX.Element => {
   const [description, setDescription] = useState<string>('');
 
   const saveData = (): void => {
-    const newData = { description };
+    const newData = { ...(settings && settings.data), description };
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply src/modules/builder/configuration/AddImageStep.tsx:46]: it going to be an empty object

File: src/modules/builder/configuration/AddLabelsStep/DraggableLabel.tsx [hunk 3]
```diff
@@ -30,14 +32,28 @@ type Props = {
...
     setIsDragging(true);
     e.stopPropagation();
+    // prevent dragging label outside the image
+    if (
+      newP.y < (ADD_LABEL_FRAME_HEIGHT - dimension.height) / 2 ||
+      newP.y >
+        (ADD_LABEL_FRAME_HEIGHT - dimension.height) / 2 + dimension.height
+    ) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply src/modules/builder/configuration/AddLabelsStep/DraggableLabel.tsx:56]: I already omitted it within the fix of x, y offset for the add labels step, So I think we can keep it for now as we are going to discard it anyway.

File: src/modules/common/ImageFrame.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+const ImageFrame = (): JSX.Element | null => {
+  const {
+    data: image,
+    isLoading: settingLoading,
+    isError,
+  } = hooks.useAppSettings({
+    name: SettingsKeys.File,
+  });
+
+  const appSettingId = image?.[0]?.id || '';
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply src/modules/common/ImageFrame.tsx:25]: unfortunately `useAppSettingFile` not accepting undefined

**Pull request merged**